spec/requests/vms_spec.rb

```diff
@@ -1416,93 +1416,7 @@ def update_raw_power_state(state, *vms)
-      content
-    end
-
-    let!(:compliance_policy) do
-      policy = FactoryBot.create(:miq_policy, :mode => "compliance", :towhat => "Vm", :description => "check_compliance", :active => true)
-      policy.conditions << FactoryBot.create(:condition)
-      policy.miq_policy_contents << policy_content
-      policy.sync_events([event_definition])
-      policy
-    end
-
-    it "to an invalid vm" do
-      api_basic_authorize action_identifier(:vms, :check_compliance)
-
-      post(invalid_vm_url, :params => gen_request(:check_compliance))
-
-      expect(response).to have_http_status(:not_found)
-    end
-
-    it "to an invalid vm without appropriate role" do
-      api_basic_authorize
-
-      post(invalid_vm_url, :params => gen_request(:check_compliance))
-
-      expect(response).to have_http_status(:forbidden)
-    end
-
-    it "to a single Vm without compliance policies assigned" do
-      api_basic_authorize action_identifier(:vms, :check_compliance)
-
-      post(vm_url, :params => gen_request(:check_compliance))
-
-      expect_single_action_result(:success => false, :message => /VM id:#{vm.id} .* has no compliance policies assigned/i, :href => api_vm_url(nil, vm))
-    end
-
-    it "to a single Vm with a compliance policy" do
-      api_basic_authorize action_identifier(:vms, :check_compliance)
-
-      vm.add_policy(compliance_policy)
-
-      post(vm_url, :params => gen_request(:check_compliance))
-
-      expect_single_action_result(:success => true, :message => /VM id:#{vm.id} .* check compliance requested/i, :href => api_vm_url(nil, vm), :task => true)
-    end
-
-    it "to multiple Vms without appropriate role" do
-      api_basic_authorize
-
-      post(api_vms_url, :params => gen_request(:check_compliance, [{"href" => vm1_url}, {"href" => vm2_url}]))
-
-      expect(response).to have_http_status(:forbidden)
-    end
-
-    it "to multiple Vms" do
-      api_basic_authorize collection_action_identifier(:vms, :check_compliance)
-
-      vm1.add_policy(compliance_policy)
-
-      post(api_vms_url, :params => gen_request(:check_compliance, [{"href" => vm1_url}, {"href" => vm2_url}]))
-
-      expected = {
-        "results" => a_collection_containing_exactly(
-          a_hash_including(
-            "message"   => a_string_matching(/VM id:#{vm1.id} .* check compliance requested/i),
-            "task_id"   => a_kind_of(String),
-            "task_href" => a_string_matching(api_tasks_url),
-            "success"   => true,
-            "href"      => api_vm_url(nil, vm1)
-          ),
-          a_hash_including(
-            "message" => a_string_matching(/VM id:#{vm2.id} .* has no compliance policies assigned/i),
-            "success" => false,
-            "href"    => api_vm_url(nil, vm2)
-          )
-        )
-      }
-
-      expect(response.parsed_body).to include(expected)
-      expect(response).to have_http_status(:ok)
-    end
-  end
+  it_behaves_like "has check compliance action", "vm", :vm_vmware, "Vm"
```

Comment:
can you rename this, sounds a bit weird in the rspec -fd output, maybe rename both to it_behaves_like "_it implements a check compliance action_"

Reply:
Now it looks like:
```
  behaves like a check compliance action
    to a single vm without compliance policies assigned
    to an invalid vm without appropriate role
    to multiple vms without appropriate role
    to a single vm with a compliance policy
    to multiple vms
    to an invalid vm
```

Reply:
👍 Thanks @lfu for the update and the refactoring, should make it easier for adding it to the other endpoints.